desktop/src/main/java/bisq/desktop/components/paymentmethods/CapitualForm.java

```diff
@@ -0,0 +1,126 @@
+                        InputValidator inputValidator, GridPane gridPane, int gridRow, CoinFormatter formatter) {
+        super(paymentAccount, accountAgeWitnessService, inputValidator, gridPane, gridRow, formatter);
+        this.capitualAccount = (CapitualAccount) paymentAccount;
+        this.capitualValidator = capitualValidator;
+    }
+
+    @Override
+    public void addFormForAddAccount() {
+        gridRowFrom = gridRow + 1;
+
+        accountNrInputTextField = FormBuilder.addInputTextField(gridPane, ++gridRow, Res.get("payment.capitual.cap"));
+        accountNrInputTextField.setValidator(capitualValidator);
+        accountNrInputTextField.textProperty().addListener((ov, oldValue, newValue) -> {
+            capitualAccount.setAccountNr(newValue);
+            updateFromInputs();
+        });
+
+        addCurrenciesGrid(true);
+        addLimitations(false);
+        addAccountNameTextFieldWithAutoFillToggleButton();
+    }
+
+    private void addCurrenciesGrid(boolean isEditable) {
+        final Tuple2<Label, FlowPane> labelFlowPaneTuple2 = addTopLabelFlowPane(gridPane, ++gridRow, Res.get("payment.supportedCurrencies"), 0);
+
+        FlowPane flowPane = labelFlowPaneTuple2.second;
+
+        if (isEditable)
+            flowPane.setId("flow-pane-checkboxes-bg");
+        else
+            flowPane.setId("flow-pane-checkboxes-non-editable-bg");
+
+        CurrencyUtil.getAllCapitualCurrencies().stream().forEach(e ->
+                fillUpFlowPaneWithCurrencies(isEditable, flowPane, e, capitualAccount));
+    }
+
+    @Override
+    protected void autoFillNameTextField() {
+        if (useCustomAccountNameToggleButton != null && !useCustomAccountNameToggleButton.isSelected()) {
+            String accountNr = accountNrInputTextField.getText();
+            accountNr = StringUtils.abbreviate(accountNr, 9);
+            String method = Res.get(paymentAccount.getPaymentMethod().getId());
+            accountNameTextField.setText(method.concat(": ").concat(accountNr));
+        }
+    }
+
+    @Override
+    public void addFormForDisplayAccount() {
+        gridRowFrom = gridRow;
+        addCompactTopLabelTextField(gridPane, gridRow, Res.get("payment.account.name"),
+                capitualAccount.getAccountName(), Layout.FIRST_ROW_AND_GROUP_DISTANCE);
+        addCompactTopLabelTextField(gridPane, ++gridRow, Res.get("shared.paymentMethod"),
+                Res.get(capitualAccount.getPaymentMethod().getId()));
+        addCompactTopLabelTextField(gridPane, ++gridRow, Res.get("payment.capitual.cap"),
+                capitualAccount.getAccountNr());
+        addLimitations(true);
+        addCurrenciesGrid(false);
+    }
+
+    @Override
+    public void updateAllInputsValid() {
+        allInputsValid.set(isAccountNameValid()
+                && capitualValidator.validate(capitualAccount.getAccountNr()).isValid
+                && capitualAccount.getTradeCurrencies().size() > 0);
```

Comment:
```suggestion
                && !capitualAccount.getTradeCurrencies().isEmpty());
```
It's faster and makes your intent clearer

Reply:
@KaiWitt Thanks for starting to do code reviews :+1: